libcaf_core/caf/load_inspector_base.hpp

```diff
@@ -106,17 +107,31 @@ class load_inspector_base : public load_inspector {
     auto tmp = value_type{};
     using setter_result = decltype(set(std::move(tmp)));
     if constexpr (std::is_same<setter_result, bool>::value) {
-      if (dref().value(tmp))
-        return set(std::move(tmp));
-      else
+      if (dref().apply(tmp)) {
+        if (set(std::move(tmp))) {
+          return true;
+        } else {
+          super::emplace_error(sec::save_callback_failed);
+          return false;
+        }
+      } else {
         return false;
+      }
+    } else if constexpr (std::is_same<setter_result, void>::value) {
+      if (dref().apply(tmp)) {
+        set(std::move(tmp));
+        return true;
+      } else {
+        return false;
+      }
     } else {
-      static_assert(std::is_convertible<setter_result, error>::value);
-      if (dref().value(tmp)) {
+      static_assert(std::is_convertible<setter_result, error>::value,
+                    "a setter must return caf::error, bool or void");
```

Comment:
Could be aligned with the message in `load_inspector.hpp` (lines: 348, 370). Granted, this is a nitpick.
```suggestion
                    "setter must return caf::error, bool or void");
```

Reply:
Yep, I like the idea of being consistent. I've updated the other places, though, because I like an actual sentence starting with "a setter" better. 🙂